spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingServiceProperties.java

```diff
@@ -127,9 +127,10 @@ public void setEnvironment(Environment environment) {
 		if (environment instanceof ConfigurableEnvironment) {
 			// override the bindings store with the environment-initializing version if in
 			// a Spring context
+			Map<String, BindingProperties> delegate = new TreeMap<String, BindingProperties>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
`bindings` can't be null here since the field is set to a default value.  The referenced `init()` method executes before this because it's a post-construct  on an autoconfiguration class and will update the map before the environment is set. 

This change ensures that any changes made to the `bindings` field before we introduce the environment-based map is preserved (which was the original issue).